### ISSUE TITLE
[BUG]  Trufflehog freezes when `error unarchiving chunk`  occurs while trying to unpack the .rar/tar files

### ISSUE THREAD

**Issue by bugbaba** [trufflesecurity/trufflehog#2390]
### TruffleHog Version
3.67.2

### Trace Output
https://gist.github.com/bugbaba/9b249a6bf308282cc5ef42940e0426ac

### Expected Behavior
Move to next part after handling the error gratefully.

### Actual Behavior
Gets stuck after the error occurs.

### Steps to Reproduce

 1. Run trufflehog against https://github.com/bugbaba/tf_eror
 2. We get the `error unarchiving chunk` error with `rardecode: bad header crc` as message.
 3. And TruffleHog just freezes and never exits.
 

## Additional Context
This was found in wild the `.rar` file is perfectly valid. Have seen the same happening with `.tar` file with below message.

```
2024-02-06T19:18:47+05:30	error	trufflehog	error unarchiving chunk.	{"source_manager_worker_id": "IPKlJ", "commit": "3ae37eb", "path": "build.tar", "timeout": 30, "error": "archive/tar: invalid tar header"}

```

**Comment by rgmz** [comment]
@mcastorina would this be related to #2388?

**Comment by mcastorina** [comment]
Yeah, that should have fixed it. I forgot to create a release 🤦 

Tested on `main` against https://github.com/bugbaba/tf_eror to confirm.

**Comment by mcastorina** [comment]
Fix is available in `v3.67.3`, thanks for reporting!

**Comment by RuchitaKshirsagarTR** [comment]
Was this issue fixed for .gz and .zip files?
